### PR TITLE
Integration testing + mongodb_admin testing

### PR DESCRIPTION
**Pull request by damienbfs**
Added integration test script and github action. Api tests can be done in integration mode (with a mongodb) or in local-only mode by switching the LOMAS_TEST_MONGO_INTEGRATION environnment variable. Added tests for user part of mongodb_admin.py file. Updated mongodb_admin.py file to add more sanity checks.

### CODE REVIEW

File: .github/workflows/test_integration_server.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Server Integration Testing
+
+on: 
+  pull_request:
+    branches: [ develop ]
```
Comment (PaulineMauryL) [inline .github/workflows/test_integration_server.yml:5]: can we add master too ?

File: server/src/constants.py [hunk 2]
```diff
@@ -2,12 +2,19 @@
 import string
 from enum import StrEnum
 
-if os.environ.get("IN_CONTAINER", False):
+# Get config and secrets from correct location
+if "LOMAS_CONFIG_PATH" in os.environ:
+    CONFIG_PATH = f"""{os.environ.get("LOMAS_CONFIG_PATH")}"""
```
Comment (PaulineMauryL) [inline server/src/constants.py:7]: why do we need 3 """ ? is that why I had my bug ?

File: server/src/tests/test_api.py [hunk 3]
```diff
@@ -36,6 +70,39 @@ def setUp(self) -> None:
...
+            create_users_collection(args)
+
+            args.path = "tests/test_data/penguin_dataset.yaml"
+            args.dataset = "PENGUIN_DATASET"
+            args.dataset_type = "LOCAL_DB"
+            args.overwrite_datasets = True
+            args.overwrite_metadata = True
+            add_datasets(args)
+
+    def tearDown(self) -> None:
```
Comment (PaulineMauryL) [inline server/src/tests/test_api.py:89]: nice ^^

File: .github/workflows/test_integration_server.yml (hunk 1 above)
Comment (damienbfs) [reply .github/workflows/test_integration_server.yml:5]: Yes, will add it.

**Review by PaulineMauryL — APPROVED** [review]

**Pull request merged**